### PR TITLE
Cast on Death implementation

### PR DESCRIPTION
**Pull request by Nostrademous**
Now DPS based Support recommendations won't include Cast on Death as top dog. If you want large average hit, it still will.

### CODE REVIEW

File: Export/Skills/sup_dex.txt [hunk 1]
```diff
@@ -109,6 +109,7 @@ local skills, mod, flag, skill = ...
 			mod("Duration", "INC", nil),
 		},
 	},
+#baseMod skill("triggeredOnDeath", true),
```
Comment (Wires77) [inline Export/Skills/sup_dex.txt:112]: #baseMod directive already adds the comma
```suggestion
#baseMod skill("triggeredOnDeath", true)
```

**Review by Wires77 — CHANGES_REQUESTED** [review]
Cast speed goes to infinity, so Cast on Death still shows up at the top of the list
image

**Review by Wires77 — APPROVED** [review]
LGTM now.  I'm tempted to wait to merge this until the next release, and we can have all of the trigger gem changes released at the same time if they're ready by then.

File: Modules/CalcPerform.lua [hunk 2]
```diff
@@ -936,6 +936,17 @@ function calcs.perform(env)
...
+		if activeSkill.skillData.triggeredOnDeath and not activeSkill.skillFlags.minion then
+			activeSkill.skillData.triggered = true
+			for i, value in ipairs(activeSkill.skillModList:Tabulate("INC", env.player.mainSkill.skillCfg, "TriggeredDamage")) do
+				activeSkill.skillModList:NewMod("Damage", "INC", value.mod.value, value.mod.source, value.mod.flags, value.mod.keywordFlags, unpack(value.mod))
+			end
+			for i, value in ipairs(activeSkill.skillModList:Tabulate("MORE", env.player.mainSkill.skillCfg, "TriggeredDamage")) do
+				activeSkill.skillModList:NewMod("Damage", "MORE", value.mod.value, value.mod.source, value.mod.flags, value.mod.keywordFlags, unpack(value.mod))
+			end
+			-- Set trigger time to 1 min in ms ( == 6000 ). Technically any large value would do.
+			activeSkill.skillData.triggerTime = 60 * 100
```
Comment (zao) [inline Modules/CalcPerform.lua:948]: What unit does `triggerTime` have? The PR seems to consistently use 100ths, but a millisecond is a 1000th of a second.

Comment (Nostrademous) [reply Modules/CalcPerform.lua:948]: right, one missing 0.

Makes no difference but I can fix it.

**Review by ppoelzl — CHANGES_REQUESTED** [review]
Waiting with merging until the fix.

**Comment by Nostrademous** [comment]
> Waiting with merging until the fix.

Fixed

**Review by ppoelzl — APPROVED** [review]
LGTM.

**Pull request merged**